### PR TITLE
Issue #1912 - add compartment search parms to front of list

### PR DESCRIPTION
**Pull request by michaelwschroeder**
Signed-off-by: Mike Schroeder <mschroed@us.ibm.com>

### CODE REVIEW

**Review by punktilious — APPROVED** [review]
LGTM

File: fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java [hunk 1]
```diff
@@ -1408,7 +1408,9 @@ public static FHIRSearchContext parseQueryParameters(String compartmentName, Str
         }
 
         FHIRSearchContext context = parseQueryParameters(resourceType, queryParameters, lenient);
-        context.getSearchParameters().addAll(parameters);
+
+        // Add the inclusion criteria search parameters to the front of the search parameter list
+        context.getSearchParameters().addAll(0, parameters);
```
Comment (lmsurpre) [inline fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java:1413]: why is parameters a list?  shouldn't it also be 0 or 1 comparment params?

**Review by lmsurpre — APPROVED** [review]
LGTM

File: fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java (hunk 1 above)
Comment (michaelwschroeder) [reply fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java]: Doesn't need to be a list - fixed.

**Pull request merged**